src/org/opendatakit/briefcase/model/BriefcasePreferences.java

```diff
@@ -21,39 +21,133 @@
+   */
+  public String get(String key, String defaultValue) {
+    return preferences.get(key, defaultValue);
+  }
+  
+  /**
+   * Associates the specified value with the specified key in this preference
+   * node.
+   * 
+   * @param key
+   *          key with which the specified value is to be associated.
+   * @param value
+   *          value to be associated with the specified key.
+   */
+  public void put(String key, String value) {
+    preferences.put(key, value);
+  }
+  
+  /**
+   * Removes the value associated with the specified key in this preference
+   * node, if any.
+   * 
+   * @param key
+   *          key whose mapping is to be removed from the preference node.
+   */
+  public void remove(String key) {
+    preferences.remove(key);
+  }
 
   public static void setBriefcaseDirectoryProperty(String value) {
-    if ( value == null ) {
-      getApplicationPreferences().remove(BRIEFCASE_DIR_PROPERTY);
+    if (value == null) {
+      Preference.APPLICATION_SCOPED.remove(BRIEFCASE_DIR_PROPERTY);
     } else {
-      getApplicationPreferences().put(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY, value);
+      Preference.APPLICATION_SCOPED.put(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY, value);
     }
   }
 
   public static String getBriefcaseDirectoryIfSet() {
-    return getApplicationPreferences().get(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY,null);
+    return Preference.APPLICATION_SCOPED.get(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY, null);
   }
 
   public static String getBriefcaseDirectoryProperty() {
-    return getApplicationPreferences().get(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY,
+    return Preference.APPLICATION_SCOPED.get(BriefcasePreferences.BRIEFCASE_DIR_PROPERTY,
         System.getProperty("user.home"));
   }
-
-  private static final String BRIEFCASE_DIR_PROPERTY = "briefcaseDir";
-
-  private static Preferences applicationPreferences = null;
-
-  private static synchronized Preferences getApplicationPreferences() {
-    if ( applicationPreferences == null ) {
-      // as good a place as any to do one-time initialization...
-      Security.addProvider(new BouncyCastleProvider());
-      // and load the preferences...
-      applicationPreferences = Preferences.userNodeForPackage(BriefcasePreferences.class);
-    }
-    return applicationPreferences;
+  
+  /**
+   * Enum that implements the strategies, to create differently scoped preferences.
+   */
+  private enum PreferenceScope {
+    APPLICATION {
+      @Override
+      public Preferences preferenceFactory(Class<?> node) {
+        return Preferences.userNodeForPackage(node);
+      }
+    },
+    CLASS_NAME {
+      @Override
+      public Preferences preferenceFactory(Class<?> node) {
+        return Preferences.userRoot().node(node.getName());
+      }
+    };
+    public abstract Preferences preferenceFactory(Class<?> node);
+  }
+  
+  /**
+   * Static member class, that holds the instance to the application scoped

```

Comment:
Thanks for writing a bit of an explanation here. I think it's still a little obscure so I'd recommend linking out to https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom so someone looking at the code can learn more about the pattern and why it works.